generators/entity/index.js

```diff
@@ -180,38 +201,48 @@ module.exports = EntityGenerator.extend({
-                this.relationships = this.fileData.relationships;
-                this.fields = this.fileData.fields;
-                this.changelogDate = this.fileData.changelogDate;
-                this.dto = this.fileData.dto;
-                this.service = this.fileData.service;
-                this.pagination = this.fileData.pagination;
-                this.javadoc = this.fileData.javadoc;
-                this.entityTableName = this.fileData.entityTableName || _s.underscored(this.name).toLowerCase();
-                this.fields && this.fields.forEach(function (field) {
-                    fieldNamesUnderscored.push(_s.underscored(field.fieldName));
-                    fieldNameChoices.push({name: field.fieldName, value: field.fieldName});
-                }, this);
-                this.relationships && this.relationships.forEach(function (rel) {
-                    relNameChoices.push({
-                        name: rel.relationshipName + ':' + rel.relationshipType,
-                        value: rel.relationshipName + ':' + rel.relationshipType
-                    });
-                }, this);
-                if (this.fileData.angularJSSuffix !== undefined) {
-                    this.entityAngularJSSuffix = this.fileData.angularJSSuffix;
-                }
+                this._loadJson();
             }
         }
     },
 
     /* private Helper methods */
+    _loadJson: function () {
+        try {
+            this.fileData = this.fs.readJSON(this.fromPath);
+        } catch (err) {
+            this.log(chalk.red('\nThe configuration file could not be read!\n'));
+            return;
+        }
+        this.relationships = this.fileData.relationships;
+        this.fields = this.fileData.fields;
+        this.changelogDate = this.fileData.changelogDate;
+        this.dto = this.fileData.dto;
+        this.service = this.fileData.service;
+        this.pagination = this.fileData.pagination;
+        this.javadoc = this.fileData.javadoc;
+        this.entityTableName = this.fileData.entityTableName || _s.underscored(this.name).toLowerCase();
+        this.fields && this.fields.forEach(function (field) {
+            fieldNamesUnderscored.push(_s.underscored(field.fieldName));
+            fieldNameChoices.push({name: field.fieldName, value: field.fieldName});
+        }, this);
+        this.relationships && this.relationships.forEach(function (rel) {
+            relNameChoices.push({name: rel.relationshipName + ':' + rel.relationshipType, value: rel.relationshipName + ':' + rel.relationshipType});
+        }, this);
+        if (this.fileData.angularJSSuffix !== undefined){
+            this.entityAngularJSSuffix = this.fileData.angularJSSuffix;
+        }
+        if (this.applicationType == 'gateway'){
+            this.microserviceName = this.fileData.microserviceName;
+            this.searchEngine = this.fileData.searchEngine;
```

Comment:
this needs to be `this.searchEngine = this.fileData.searchEngine || this.searchEngine` to handle all cases